Add project downloads section

allyhawkins opened the pull request:
Closes #391 

Here I am starting the changes to the `Downloadable files` page to reflect the new [IA](https://data-lab-knowledge.slab.com/posts/sc-pca-docs-ia-changes-for-cart-0y0xuuc0). Since this is the first PR where I'm starting to change things around, I also included some general reorganization of headers to reflect the IA and added in TODOs for sections that will need to be filled out. 

I mostly kept the intro the same, but added in some sentences mentioning there are three ways to download things - picking a single project, making a custom dataset, and downloading one of the portal-wide options. Here I linked to each of the three sections that will talk about these downloads in more detail. 

I filled in the project download section, which will include the screenshots for downloading a project as either SCE or AnnData objects. I moved the description of the bulk files from the intro to this section, since they are only applicable when downloading an entire project. 

I then moved the section for merged objects to be nested within the project download section and made some small edits. 

We will need to update the images for project downloads, which is tracked in #382. I'll add a comment there with the list of which images will need to be regenerated. I do have one TODO, about whether or not we should show an example project download structure for AnnData with CITE-seq? I think yes?

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
+## Project downloads
 
-See also {ref}`processing bulk RNA samples <processing_information:Bulk RNA samples>`.
+To download the gene expression data for all samples in a single project, use the `Download Now` button next to the project title. 
+This will allow you to download the chosen project instantly, without having to wait for a download link to be available. 
```
I'm not sure we need all this detail about the link, since the concept of having to wait for a download link hasn't introduced in the first place, so I'm suggesting to condense:

```suggestion
Use the `Download Now` button next to the project title to instantly download gene expression data for all samples in a single project.
```

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
+For more information on choosing a data format and modality, see the {ref}`section on download options<STUB TO DATA FORMATS SECTION>`.
 
-## `SingleCellExperiment` downloads
+If downloading a project containing bulk RNA-seq data, two tab-separated value files, e.g., `SCPCP000000_bulk_quant.tsv` and `SCPCP000000_bulk_metadata.tsv`, will be included in the project download.
```
```suggestion
If the project contains bulk RNA-seq data, two tab-separated value files, e.g., `SCPCP000000_bulk_quant.tsv` and `SCPCP000000_bulk_metadata.tsv`, will also be included in the project download.
```

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
 
-## `SingleCellExperiment` downloads
+If downloading a project containing bulk RNA-seq data, two tab-separated value files, e.g., `SCPCP000000_bulk_quant.tsv` and `SCPCP000000_bulk_metadata.tsv`, will be included in the project download.
+The `SCPCP000000_bulk_quant.tsv` file contains a gene by sample matrix (each row a gene, each column a sample) containing raw gene expression counts quantified by Salmon.
```
```suggestion
The `SCPCP000000_bulk_quant.tsv` file contains a gene by sample matrix (each row a gene, each column a sample) containing raw gene expression counts quantified by `salmon`.
```

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
+![project download folder](images/anndata-project-download-folder.png){width="600"}
 
-## `AnnData` downloads
+<!--TODO: Add in project download showing at least one sample with CITE-seq?-->
```
Yeah, I think we do want this

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
+### Metadata-only downloads
+
+<!--Update this text once adding in the intro for portal wide downloads-->
+A single TSV file containing the metadata for all samples from all projects on the Portal is available for download.
```
```suggestion
A single TSV file containing the metadata for all single-cell and single-nuclei samples from all projects on the Portal is available for download.
```

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,109 @@ The files shown below will be included with each library (example shown for a li
 
 If downloading a sample that contains a CITE-seq library as an `AnnData` object (`.h5ad` file), the quantified CITE-seq expression data is included as a separate file with the suffix `_adt.h5ad`.
 
+<!--TODO: 
+move spatial section 
+merged objects
+-->
```
this seems done, yeah?

```suggestion
```

allyhawkins replied:
No, we haven't moved the spatial section since it's still at the bottom. And then we will need to mention merged objects specifically for the portal-wide downloads. These TODOs are tracked in #395 and #396.

sjspielman replied:
Ok, I got a little thrown off with all the diffs, I was reading this to imply these were moved _away_ from here. Eh?

```suggestion
<!--TODO: 
spatial section to be moved here, followed by the merged objects section
-->
```

sjspielman reviewed (commented):
I think this looks good! Thanks for taking care of a lot of the organizational items at this stage too; will certainly help us avoid conflicts working on this same file across branches 😬

I left a couple comments but nothing major! The main other thought I have is, maybe we want an issue as a placeholder to make sure we get all the TODOs and STUBs before heading to prod.

sjspielman commented on docs/download_files.md:
```diff
@@ -2,11 +2,12 @@
 We recommend you record this date in case there are future updates to the Portal that change the underlying data or if you need to cite the data in the future (see {ref}`How to Cite <citation:how to cite>` for more information).
 Please see our {ref}`CHANGELOG <CHANGELOG:CHANGELOG>` for a summary of changes that impact downloads from the Portal.
 
+Data can be downloaded by either downloading a [single project](#project-downloads), creating a [custom dataset](#custom-datasets), or by choosing one of the [portal-wide download options](#portal-wide-downloads).
```
Noting you can also download single samples

allyhawkins replied:
With the addition of cart, you can't just download a single sample anymore. You will have to add it to my dataset and then download. All of that will be described in the "custom dataset" section.

allyhawkins commented:
@sjspielman I believe I addressed all your comments or responded where appropriate. Just a note that I didn't directly accept your suggestion here https://github.com/AlexsLemonade/scpca-docs/pull/411#discussion_r2077560081, but instead made some edits to clarify what's actually included in the portal wide metadata download. I think this can now mean we close #392 once this goes in. 

> I think this looks good! Thanks for taking care of a lot of the organizational items at this stage too; will certainly help us avoid conflicts working on this same file across branches 😬

Just a minor note that the docs you are working on for download options, should be a separate file!

sjspielman commented on docs/download_files.md:
```diff
@@ -21,42 +22,107 @@ The files shown below will be included with each library (example shown for a li
+
 ## Gene expression data
 
 Single-cell or single-nuclei gene expression data is provided as either [`SingleCellExperiment` objects (`.rds` files)](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html) or [`AnnData` objects (`.h5ad` files)](https://anndata.readthedocs.io/en/latest/index.html).
```
i just noticed this while scrolling through, do we want to update any of these OSCA links (more generally?)

sjspielman reviewed (approved):
LGTM!
